front/app/components/UI/Loading/index.tsx

```diff
@@ -0,0 +1,17 @@
+import React, { Suspense } from 'react';
+import FullPageSpinner from '../FullPageSpinner';
+
+interface Props {
+  // If we're in the admin, the loader will take into account
+  // the admin navbar's width.
+  admin?: boolean;
+  children: React.ReactNode;
+}
+
+const Loading = ({ admin = false, children }: Props) => {
```

Comment:
Perhaps we can come up with a more descriptive name for this component as we don't want people to use it for anything but routes, correct? Maybe `PageLoading` or if you can come up with something better?